nowcasting_dataset/data_sources/datasource_output.py

```diff
@@ -0,0 +1,155 @@
+""" General Data Source output pydantic class. """
+from pydantic import BaseModel, Field
+import pandas as pd
+import xarray as xr
+import numpy as np
+from typing import List
+import logging
+
+from nowcasting_dataset.utils import to_numpy
+
+logger = logging.getLogger(__name__)
+
+
+class DataSourceOutput(BaseModel):
+    """General Data Source output pydantic class.
+
+    Data source output classes should build of this class
+    """
+
+    class Config:
+        """ Allowed classed i.e. tensor.Tensor"""
+
+        # TODO maybe there is a better way to do this
+        arbitrary_types_allowed = True
+
+    batch_size: int = Field(
+        0,
+        ge=0,
+        description="The size of this batch. If the batch size is 0, "
+        "then this item stores one data item",
+    )
+
+    def to_numpy(self):
+        """Change to numpy"""
+        for k, v in self.dict().items():
+            self.__setattr__(k, to_numpy(v))
+
+    def to_xr_data_array(self):
+        """ Change to xr DataArray"""
+        pass
```

Comment:
Assuming we want sub-classes to define this function, then I'd do:

```suggestion
        raise NotImplementedError()
```

Unless we really do want `to_xr_data_array()` to silently do nothing??